### PR TITLE
[JuPedSim] Improved FCD output.

### PR DESCRIPTION
**Pull request by bcoueraud87**

### CODE REVIEW

File: src/microsim/transportables/MSPModel_JuPedSim.h [hunk 1]
```diff
@@ -80,26 +80,33 @@ class MSPModel_JuPedSim : public MSPModel {
         Position getPosition(const MSStageMoving& stage, SUMOTime now) const;
         void setPosition(double x, double y);
 
+        Position getPreviousPosition() const;
```
Comment (behrisch) [inline src/microsim/transportables/MSPModel_JuPedSim.h:83]: This function gets never called

File: src/microsim/transportables/MSPModel_JuPedSim.h [hunk 2]
```diff
@@ -114,7 +121,10 @@ class MSPModel_JuPedSim : public MSPModel {
         Position myDestination;
         JPS_AgentId myAgentId;
         Position myPosition;
+        Position myPreviousPosition; // Will be initialized to zero automatically.
         double myAngle;
+        double myLanePosition;
+        MSEdge* myCurrentEdge;
```
Comment (behrisch) [inline src/microsim/transportables/MSPModel_JuPedSim.h:127]: This seems to be unused

File: src/microsim/transportables/MSPModel_JuPedSim.cpp [hunk 3]
```diff
@@ -696,12 +723,13 @@ SUMOTime MSPModel_JuPedSim::PState::getWaitingTime(const MSStageMoving& /* stage
 
 
 double MSPModel_JuPedSim::PState::getSpeed(const MSStageMoving& /* stage */) const {
-    return 0;
+    Position velocity = (myPosition - myPreviousPosition) / STEPS2TIME(DELTA_T);
+    return velocity.length2D();
 }
```
Comment (behrisch) [inline src/microsim/transportables/MSPModel_JuPedSim.cpp:728]: This is quite inefficient. You could just use myPosition.distanceTo2D(myPreviousPosition) / TS

**Review by behrisch — CHANGES_REQUESTED** [review]

File: src/microsim/transportables/MSPModel_JuPedSim.h (hunk 1 above)
Comment (bcoueraud87) [reply src/microsim/transportables/MSPModel_JuPedSim.h:83]: Yes. I removed it, there is no need to expose this attribute.

File: src/microsim/transportables/MSPModel_JuPedSim.h (hunk 2 above)
Comment (bcoueraud87) [reply src/microsim/transportables/MSPModel_JuPedSim.h]: `myCurrentEdge` has been removed, thanks.

**Pull request merged**